ika2btc/processor.go

```diff
@@ -0,0 +1,132 @@
+package ika2btc
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/gonative-cc/relayer/bitcoin"
+	"github.com/gonative-cc/relayer/dal"
+	"github.com/rs/zerolog/log"
+)
+
+// Processor handles processing transactions from IKA to Bitcoin.
+type Processor struct {
+	BtcClient               bitcoin.Client
+	db                      *dal.DB
+	txConfirmationThreshold uint8
+}
+
+// NewProcessor creates a new Processor instance.
+func NewProcessor(
+	btcClientConfig rpcclient.ConnConfig,
+	confirmationThreshold uint8,
+	db *dal.DB,
+) (*Processor, error) {
+
+	if db == nil {
+		err := fmt.Errorf("database cannot be nil")
+		log.Err(err).Msg("")
+		return nil, err
+	}
+
+	if btcClientConfig.Host == "" || btcClientConfig.User == "" || btcClientConfig.Pass == "" {
+		err := fmt.Errorf("missing bitcoin node configuration")
+		log.Err(err).Msg("")
+		return nil, err
+	}
+
+	client, err := rpcclient.New(&btcClientConfig, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Processor{
+		BtcClient:               client,
+		db:                      db,
+		txConfirmationThreshold: confirmationThreshold,
+	}, nil
+}
+
+// ProcessSignedTxs processes signed transactions from the database.
+func (p *Processor) ProcessSignedTxs(mu *sync.Mutex) error {
+	mu.Lock()
+	defer mu.Unlock()
```

Comment:
let's remove mutex now, we will do it in a different way.

Reply:
Done